File: examples/tentacles/hello-world/ruby/handler.rb
require "json"
raw = STDIN.read
raw = "{}" if raw.nil? || raw.empty?
req = JSON.parse(raw)
name = (((req["payload"] || {})["name"]) || "world")
print({ v: 1, id: req["id"], ok: true, result: { arm: "hello-rb", lang: "ruby", hello: "hello, #{name}" } }.to_json)

File: examples/tentacles/hello-world/perl/handler.pl
#!/usr/bin/env perl
use strict;
use warnings;
use JSON::PP;
my $raw = do { local $/; <STDIN> };
$raw = '{}' unless defined $raw && length $raw;
my $req = decode_json($raw);
my $name = ($req->{payload}{name} // 'world');
print encode_json({ v => 1, id => $req->{id}, ok => JSON::PP::true, result => { arm => 'hello-pl', lang => 'perl', hello => "hello, $name" } });
